public/app/features/dashboard-scene/scene/ScopesFiltersScene.tsx

```diff
@@ -9,111 +10,269 @@ import {
-      value = undefined;
+    if (selectedIdx === -1) {
+      const scope = await this.scopesServer.get(linkId);
+
+      if (scope) {
+        scopes = [...scopes, scope];
+      }
+    } else {
+      scopes.splice(selectedIdx, 1);
     }
 
-    this.setState({ scopes, pendingValue: undefined, value });
+    this.setState({ scopes });
   }
 }
 
 export function ScopesFiltersSceneRenderer({ model }: SceneComponentProps<ScopesFiltersScene>) {
-  const { scopes, isLoading, value } = model.useState();
+  const { nodes, expandedNodes, scopes } = model.useState();
   const parentState = model.parent!.useState();
   const isViewing = 'isViewing' in parentState ? !!parentState.isViewing : false;
 
-  const options: Array<SelectableValue<string>> = scopes.map(({ metadata: { name }, spec: { title, category } }) => ({
-    label: title,
-    value: name,
-    description: category,
-  }));
+  return (
+    <Toggletip
+      content={
+        <ScopesTreeLevel
+          isExpanded
+          path={[]}
+          nodes={nodes}
+          expandedNodes={expandedNodes}
+          scopes={scopes}
+          model={model}
+        />
+      }
+      footer={'Open advanced scope selector'}
+      closeButton={false}
+    >
+      <Input disabled={isViewing} readOnly value={scopes.map((scope) => scope.spec.title)} />
+    </Toggletip>
+  );
+}
+
+export interface ScopesTreeLevelProps {
+  isExpanded: boolean;
+  path: string[];
+  nodes: Record<string, Node>;
+  expandedNodes: string[];
+  scopes: Scope[];
+  model: ScopesFiltersScene;
+}
+
+export function ScopesTreeLevel({ isExpanded, path, nodes, expandedNodes, scopes, model }: ScopesTreeLevelProps) {
+  const styles = useStyles2(getStyles);
+
+  if (!isExpanded) {
+    return null;
+  }
 
   return (
-    <Select
-      isClearable
-      isLoading={isLoading}
-      disabled={isViewing}
-      options={options}
-      value={value}
-      onChange={(selectableValue) => model.setScope(selectableValue?.value ?? undefined)}
-    />
+    <div role="tree" className={path.length > 0 ? styles.innerLevelContainer : undefined}>
+      {Object.values(nodes).map((node) => {
+        const {
+          item: { nodeId, linkId },
+          isScope,
+          children,
+        } = node;
+        const nodePath = [...path, nodeId];
+        const isExpanded = expandedNodes.includes(nodeId);
+        const isSelected = isScope && !!scopes.find((scope) => scope.metadata.name === linkId);
+
+        return (
+          <div
+            key={nodeId}
+            role="treeitem"
+            aria-selected={isExpanded}
+            tabIndex={0}
+            className={cx(styles.item, isScope && styles.itemScope)}
+            onClick={(evt) => {
+              evt.stopPropagation();
+              model.expandNode(nodePath);
+            }}
+            onKeyDown={(evt) => {
+              evt.stopPropagation();
+              model.expandNode(nodePath);
+            }}
```

Comment:
Think we need to be more thoughtful here, currently the API is being hit very eagerly. For example, I have my focus (after tabbing into it) on the first node (Applications):

<img width="265" alt="image" src="https://github.com/grafana/grafana/assets/2376619/d2d81d7d-b168-4b36-afef-05942f48fc7e">

I would expect I can navigate using the keyboard, but instead I'm stuck on the Application node and pressing arrow down just re-executes the same API (`http://localhost:3000/apis/scope.grafana.app/v0alpha1/namespaces/default/find?parent=applications`)

Reply:
yeah, understand that!